tests/tasks/firebolt/test_firebolt.py

```diff
@@ -0,0 +1,134 @@
+                username="test",
+                password="test",
+                query="test",
+            )
+
+        # raises Value error if database name is not provided
+        with pytest.raises(ValueError, match="A database name must be provided"):
+            FireboltQuery().run(
+                username="test",
+                password="test",
+                engine_name="test",
+                query="test",
+            )
+        # raises Value error if username is not provided
+        with pytest.raises(ValueError, match="User name must be provided"):
+            FireboltQuery().run(
+                database="test",
+                password="test",
+                engine_name="test",
+                query="test",
+            )
+
+        # raises Value error if password is not provided
+        with pytest.raises(ValueError, match="A password must be provided"):
+            FireboltQuery().run(
+                database="test",
+                username="test",
+                engine_name="test",
+                query="test",
+            )
+
+        # raises Value error if query is not provided
+        with pytest.raises(ValueError, match="A query string must be provided"):
+            FireboltQuery().run(
+                database="test",
+                username="test",
+                password="test",
+                engine_name="test",
+            )
+
+    # test to check if the ddl/dml query was executed
+    def test_execute_query(self, mock_conn):
+        """
+        Tests that the FireboltQuery Task calls the execute method on the cursor.
+        """
+
+        # setting execute return
+        mock_conn.return_value.__enter__.return_value.execute.return_value = 0
+        query = "SHOW DATABASES"
+        output = FireboltQuery(
+            database="test",
+            username="test",
+            password="test",
+            engine_name="test",
+            query=query,
+        ).run()
+        mock_conn.assert_called_with()
+        mock_conn.return_value.__enter__.return_value.execute.assert_called_once_with(
+            query
+        )
+        mock_conn.return_value.__enter__.return_value.fetchall.assert_not_called()
+        assert output == []
+
+    # test to check if the query was executed and metadata was retrieved from database
+    def test_execute_fetchall(self, mock_conn):
+        """
+        Tests that the FireboltQuery Task calls the fetchall method on the cursor.
+        """
+
+        # setting fetchall return
+        mock_conn.return_value.__enter__.return_value.execute.return_value = 1
+        mock_conn.return_value.__enter__.return_value.fetchall.return_value = ["TESTDB"]
+        query = "SHOW DATABASES"
+        output = FireboltQuery(
+            database="test",
+            username="test",
+            password="test",
+            engine_name="test",
+            query=query,
+        ).run()
+        mock_conn.assert_called_with()
```

Comment:
I think this assertion should check to make sure that the `connect` method was called with the expected arguments (`database`, `username`, etc.)